source/d3d11/d3d11_device_context.cpp

```diff
@@ -421,8 +431,15 @@ void    STDMETHODCALLTYPE D3D11DeviceContext::ClearUnorderedAccessViewFloat(ID3D
 }
 void    STDMETHODCALLTYPE D3D11DeviceContext::ClearDepthStencilView(ID3D11DepthStencilView *pDepthStencilView, UINT ClearFlags, FLOAT Depth, UINT8 Stencil)
 {
+	if (_device->_runtimes.empty())
+		return;
+
+	const auto runtime = _device->_runtimes.front();
+
+	runtime->on_clear_depthstencil_view(pDepthStencilView);
+
 #if RESHADE_DX11_CAPTURE_DEPTH_BUFFERS
-	if (ClearFlags & D3D11_CLEAR_DEPTH)
+	if (ClearFlags & D3D11_CLEAR_DEPTH || (runtime->depth_buffer_more_copies && ClearFlags & D3D11_CLEAR_STENCIL))
```

Comment:
I see little point in tracking stencil clears since we are only interested in depth?

Reply:
Me too, but it makes Bioshock Infinite depth buffer detection work... As it is in very few games, i prefer to let Reshade filter by default on D3D11_CLEAR_DEPTH, then, if necessary, filter also on D3D11_CLEAR_STENCIL...